intellij/src/saros/intellij/editor/colorstorage/ColorManager.java

```diff
@@ -1,44 +1,168 @@
 package saros.intellij.editor.colorstorage;
 
-import java.awt.Color;
-
-/** IntelliJ color manager */
-// todo: temporary implementation to provide default colors
-public class ColorManager {
-
-  public static final Color DEFAULT_COLOR = new Color(128, 128, 128);
-
-  static final Color[] CONTRIBUTION_COLORS = {
-    new Color(141, 206, 231),
-    new Color(191, 187, 130),
-    new Color(186, 220, 81),
-    new Color(237, 237, 169),
-    new Color(137, 180, 178)
-  };
-
-  static final Color[] SELECTION_COLORS = {
-    new Color(183, 224, 240),
-    new Color(208, 205, 164),
-    new Color(220, 237, 166),
-    new Color(246, 246, 211),
-    new Color(184, 210, 209)
-  };
+import com.google.common.collect.ImmutableList;
+import com.intellij.openapi.editor.HighlighterColors;
+import com.intellij.openapi.editor.colors.TextAttributesKey;
+import com.intellij.openapi.editor.markup.TextAttributes;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * IntelliJ color manager. This specifies the available colors that are used for Saros highlights
+ * and that are available for configuration inside the Color Scheme preferences.
+ */
+public final class ColorManager {
+
+  /**
+   * Key for the default selection color field. This value must match the option name for the
+   * default selection color used in the color config files located in {@link colorSchemes}.
+   */
+  private static final String DEFAULT_SELECTION_KEY = "SAROS_DEFAULT_TEXT_SELECTION";
+  /**
+   * Key for the default contribution color field. This value must match the option name for the
+   * default contribution color used in the color config files located in {@link colorSchemes}.
+   */
+  private static final String DEFAULT_CONTRIBUTION_KEY = "SAROS_DEFAULT_TEXT_CONTRIBUTION";
+  /**
+   * Prefix for the user selection color field. This value must match the prefix of the option names
+   * for the user selection color used in the color config files located in {@link colorSchemes}.
+   */
+  private static final String SELECTION_KEY_PREFIX = "SAROS_TEXT_SELECTION_";
+  /**
+   * Prefix for the user contribution color field. This value must match the prefix of the option
+   * names for the user contribution color used in the color config files located in {@link
+   * colorSchemes}.
+   */
+  private static final String CONTRIBUTION_KEY_PREFIX = "SAROS_TEXT_CONTRIBUTION_";
+
+  /** Number of supported users. */
+  private static final int USER_COUNT = 5;
+
+  /** Color keys for the default colors. Used if no color keys are available for a given id. */
+  public static final ColorKeys DEFAULT_COLOR_KEYS = new DefaultColorKeys();
+  /**
+   * List of keys for supported colors. A user can select one of these colors by referencing the
+   * {@link IdentifiableColorKeys#getId()}.
+   */
+  public static final List<IdentifiableColorKeys> COLOR_KEYS;
+
+  static {
+    final ImmutableList.Builder<IdentifiableColorKeys> builder = ImmutableList.builder();
```

Comment:
Do you use Google Collection at all ? I personally do not find it a good behavior to use multiple collection frameworks.

Reply:
I did not write this part of the code. It is part of the code contributed in #482. But, it did not bother me enough to change it.

Reply:
I could replace this by just using normal collections (lists, maps) and then using `Collections.unmodifiableList(...)` or `Collections.unmodifiableMap(...)` to create the same outcome. (Even though unmodifiable is not the same as immutable in general, it should hold in this case as there is no reference to the internal representation of the unmodifiable collection, which *should* mean that it can't be modified by anything.)

@srossbach Would this be a preferable solution?

Reply:
If it does not consume too much time to change it, why not. Have seen the usage also in another class.

Reply:
Yes, in `SarosColorsPage`. Should only take ~5 minutes. Will do.

Reply:
Done.